### PR TITLE
Uml 380 actor account creation feature test

### PR DESCRIPTION
**Pull request by SeemaMenon**

### CODE REVIEW

File: service-api/app/features/actor-account-creation.feature [hunk 1]
```diff
@@ -0,0 +1,31 @@
+@actor @accountcreation
+Feature: Account creation
+  As a new user
+  I want to create an account
+  So that I can login to add my lpas to share
+
+  @integration @acceptance
+  Scenario: As a new user want to create an account
```
Comment (cooperaj) [inline service-api/app/features/actor-account-creation.feature]: A new user can create an account

File: service-api/app/features/actor-account-creation.feature [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    Given I am not a user of the lpa application
+    And I want to create a new account
+    When I create an account
+    Then I receive unique instructions on how to activate my account
+
+  @integration @acceptance
+  Scenario: The user can follow their unique instructions to activate new account
```
Comment (cooperaj) [inline service-api/app/features/actor-account-creation.feature]: The user can follow their unique instructions to activate a new account

File: service-api/app/features/actor-account-creation.feature [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    Given I have asked for creating new account
```
Comment (cooperaj) [inline service-api/app/features/actor-account-creation.feature]: I have asked to create a new account
(repeated below)

File: service-front/app/features/actor-account-creation.feature [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  So that I can login to add my lpas to share
+
+  @ui @integration
+  Scenario: As a new user want to create an account
+    Given I am not a user of the lpa application
+    And I want to create a new account
+    When I create an account
+    Then I receive unique instructions on how to create an account
+
+  @ui
```
Comment (cooperaj) [inline service-front/app/features/actor-account-creation.feature]: No integration?

File: service-front/app/features/actor-account-creation.feature [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+     And I want to create a new account
+     When I create an account using duplicate details
+     Then I receive unique instructions on how to create an account
+
+  @ui @integration
+  Scenario Outline: As a new user I want to be shown the mistakes I make while creating an account
+    Given I am not a user of the lpa application
+    And I want to create a new account
+    When I have not provided required email and password for account creation
+    Then I should be shown account cannot be created due to the below <reasons>
```
Comment (cooperaj) [inline service-front/app/features/actor-account-creation.feature]: I should be told my account could not be created due to \<reasons\>

File: service-front/app/features/context/Integration/AccountContext.php [hunk 6]
```diff
@@ -211,4 +213,201 @@ public function iAmToldThatMyPasswordIsInvalidBecauseItNeedsAtLeast($reason)
...
+                    StatusCodeInterface::STATUS_OK,
+                    [],
+                    json_encode([])
+                )
+            );
+
+        $userData = $this->userService->create('', '');
+
+        assertEmpty($userData);
+        //assertEquals($this->activationToken, $userData['activationToken']);
```
Comment (cooperaj) [inline service-front/app/features/context/Integration/AccountContext.php]: comment

File: service-front/app/features/actor-account-creation.feature [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    Then I am told my unique instructions to activate my account have expired
+
+   @ui
+   Scenario: The user account creates an account which already exists
+     Given I am not a user of the lpa application
+     And I want to create a new account
+     When I create an account using duplicate details
+     Then I receive unique instructions on how to create an account
+
+  @ui @integration
```
Comment (cooperaj) [inline service-front/app/features/actor-account-creation.feature]: This one and the others probably don't need integration tests as the validation is done in the framework. You'd only need integration testing of parts that touch the service layer (duplicate checking)

File: service-front/app/features/actor-account-creation.feature [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    And I want to create a new account
+    When I provide mismatching emails and passwords when creating account
+    Then I should be shown account cannot be created due to the below <reasons>
+    Examples:
+      | reasons                          |
+      | The emails did not match         |
+      | The passwords did not match      |
+
+  @ui @integration
+  Scenario Outline: As a new user I want to be shown the mistakes I make while creating an account
```
Comment (cooperaj) [inline service-front/app/features/actor-account-creation.feature]: It is probably a good idea to combine all three of these scenarios into a single one. You can provide a full table of data that would make up the test fixtures.

| email1 | email2 | password1 | password2 | terms | reason |
| | | | | | Enter your email address |
| a@b.com | c@d.com | test | test | yes | The emails did not match |

Just a thought

File: service-front/app/features/context/UI/AccountContext.php [hunk 9]
```diff
@@ -41,13 +45,198 @@ public function seedFixtures()
...
+
+    /**
+     * @Given /^I have asked for creating new account$/
+     */
+    public function iHaveAskedForCreatingNewAccount()
+    {
+        $this->email = 'test@example.com';
+        $this->password = 'n3wPassWord';
+        $this->activationToken = 'activate1234567890';
+        $this->lastLogin = new DateTime('now');
```
Comment (cooperaj) [inline service-front/app/features/context/UI/AccountContext.php]: This isn't set during account creation, just login

File: service-api/app/features/actor-account-creation.feature (hunk 1 above)
Comment (SeemaMenon) [reply service-api/app/features/actor-account-creation.feature]: Corrected

File: service-api/app/features/actor-account-creation.feature (hunk 2 above)
Comment (SeemaMenon) [reply service-api/app/features/actor-account-creation.feature]: Corrected

File: service-api/app/features/actor-account-creation.feature (hunk 3 above)
Comment (SeemaMenon) [reply service-api/app/features/actor-account-creation.feature]: Corrected all.

File: service-front/app/features/context/Integration/AccountContext.php (hunk 6 above)
Comment (SeemaMenon) [reply service-front/app/features/context/Integration/AccountContext.php]: Removed, sorry.

File: service-front/app/features/actor-account-creation.feature (hunk 5 above)
Comment (SeemaMenon) [reply service-front/app/features/actor-account-creation.feature]: Corrected

File: service-front/app/features/actor-account-creation.feature (hunk 7 above)
Comment (SeemaMenon) [reply service-front/app/features/actor-account-creation.feature]: Removed

File: service-front/app/features/context/UI/AccountContext.php (hunk 9 above)
Comment (SeemaMenon) [reply service-front/app/features/context/UI/AccountContext.php]: Removed

**Comment by SeemaMenon** [comment]
This branch has unwanted commits. Hence closing this PR and opening a new one for the ticket UML-380
